device_doctor/lib/src/ios_device.dart

```diff
@@ -127,7 +127,46 @@ class IosDevice implements Device {
 
   @override
   Future<void> recover() async {
-    // Restarts the device first.
-    await eval('idevicediagnostics', <String>['restart']);
+    await restart_device();
+    await uninstall_applications();
+  }
+
+  /// Restart iOS device.
+  @visibleForTesting
+  Future<bool> restart_device({ProcessManager processManager}) async {
+    processManager ??= LocalProcessManager();
+    try {
+      await eval('idevicediagnostics', <String>['restart'], processManager: processManager);
+    } on BuildFailedError catch (error) {
+      logger.severe('device restart fails: $error');
```

Comment:
should we also sdterr or is it included on the exception?

Reply:
Added stderr as well.